pkg/channel/distributed/controller/kafkachannel/reconciler.go

```diff
@@ -206,17 +214,37 @@ func (r *Reconciler) reconcile(ctx context.Context, channel *kafkav1beta1.KafkaC
 	// ConfigMap.  Therefore, we will instead check the Kafka Secret associated with the
 	// KafkaChannel here.
 	//
-	if len(r.kafkaSecret) > 0 {
+	if len(r.config.Kafka.AuthSecretName) > 0 {
 		channel.Status.MarkConfigTrue()
 	} else {
 		channel.Status.MarkConfigFailed(event.KafkaSecretReconciled.String(), "No Kafka Secret For KafkaChannel")
 		return fmt.Errorf(constants.ReconciliationFailedError)
 	}
 
+	// Reconcile the Receiver Deployment/Service
+	secret, err := r.kubeClientset.CoreV1().Secrets(r.config.Kafka.AuthSecretNamespace).Get(ctx, r.config.Kafka.AuthSecretName, metav1.GetOptions{})
+	if err != nil && apierrs.IsNotFound(err) {
+		// The Receiver reconciler needs the namespace and name for various purposes, so construct a Secret with the required values
+		err = r.reconcileReceiver(ctx, channel, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: r.config.Kafka.AuthSecretName, Namespace: r.config.Kafka.AuthSecretNamespace}}, false)
+		if err != nil {
+			logging.FromContext(ctx).Error("Reconcile Receiver With Empty Secret Failed", zap.Error(err))
+		}
+
+		// Not having an auth secret for the Receiver is a problem
+		channel.Status.MarkConfigFailed(event.KafkaSecretReconciled.String(), "No Kafka Secret For KafkaChannel")
+		return fmt.Errorf(constants.ReconciliationFailedError)
+	} else if err != nil {
+		logging.FromContext(ctx).Error("Error reading receiver secret", zap.Error(err))
+		return fmt.Errorf(constants.ReconciliationFailedError)
+	}
+
```

Comment:
I would vote (unless theres a good reason not to?) that the above logic be refactored INTO the `reconcileReceiver()` function such that this top-level "reconciler" function can just ALWAYS call `reconcileReceiver()` which will check for secrets and such.  would make this function more "consistent" and keep the "receiver" specific logic in the receiver.go file, etc..

Reply:
That's a very good idea; it saved a bit of trouble as well as being better organized